File: Geek hates too many 1s - GFG/geek-hates-too-many-1s.py
class Solution:
    def noConseBits(self, n : int) -> int:
        # code here
        s = list(bin(n)[2:])
        cnt = 0
        for i in range(len(s)):
            if s[i] == "1":
                if cnt == 2:
                    s[i] = "0"
                    cnt = 0
                else:
                    cnt += 1
            else:
                cnt = 0
        return int("".join(s),2)
            
            



#{ 
 # Driver Code Starts
if __name__=="__main__":
    t = int(input())
    for _ in range(t):
        
        n = int(input())
        
        obj = Solution()
        res = obj.noConseBits(n)
        
        print(res)
        

# } Driver Code Ends

File: Geek hates too many 1s - GFG/geek-hates-too-many-1s.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {

  public:
    int noConseBits(int n) {
        // code here
        int i = 31, count = 0;
        while(i >= 0) {
            if(count == 2 && (n & (1 << i))) n = (n & (~ (1 << i))), count = 0;
            else if((n & (1 << i))) count++;
            else count = 0;
            i--;
        }
        return n;
    }
};


//{ Driver Code Starts.

int main() {

    int tt;
    cin >> tt;
    Solution sol;
    while (tt--) {

        int n;
        cin >> n;
        int ans = sol.noConseBits(n);
        cout << ans << '\n';
    }

    return 0;
}

// } Driver Code Ends